src/fileutil.cc

```diff
@@ -159,50 +159,42 @@ namespace {
 class GlobCache {
  public:
   ~GlobCache() { Clear(); }
-
-  void Get(const char* pat, vector<string>** files) {
-    auto p = cache_.emplace(pat, nullptr);
-    if (p.second) {
-      vector<string>* files = p.first->second = new vector<string>;
+  const GlobMap::mapped_type& Get(const char* pat) {
+    auto [it, inserted] = cache_.try_emplace(pat);
+    auto& files = it->second;
```

Comment:
Mind making this `vector<string>` instead? The type of `cache_` is complicated enough that it's not quite obvious what the type of `files` is.

Reply:
Though I have a preference to only have to specify the type once, I will make this change :-)

(in fact there are more occurrences where we could deduct the type instead without loss of context.)

Reply:
Hmm, while looking at the code, my preference got stronger. The way it is done right now is that there is a single definition of the map type (`GlobMap`) and all signatures and such are derived from that type through standard type aliases. It would be odd to not do it in this single place. (Also, this is the implementation detail that is right next to the type definition. Not hard to find out the actual type.) Adding full types everywhere would be quite verbose and definitely not [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). So, what do you think?

Reply:
Dunno, I don't have a lot of experience with type inference and large code bases so I wouldn't be speaking with any sort of authority; my purely personal opinion is that it's nice to know what the type of `files` is without going spelunking in the source code. I suppose a good IDE could help, but do C++ IDEs help with this?

In any case, it's at most a minor issue so meh.